### ISSUE TITLE
[rllib, tune] Passing tunable hyper parameter into custom model

### ISSUE THREAD

**Issue by pengmun** [ray-project/ray#7456]
I am trying to set up the dropout as a tunable hyperparameter with PBT using the default PPO agent with a custom model.

I was able to set up the custom model based on the custom Keras model example, however I am having difficulties accessing the current config from within the model. Based on the examples with dropout and Tune, they uses self.config.get("dropout") to obtain the current config value for dropout hyperparameter. However the config is accessible to the trainable but is not passed to the model.

Is there an easy way to access the config dropout value from within the custom model?

**Comment by richardliaw** [comment]
I think you need to specify it as part of `model_config`.

cc @sven1977

**Comment by pengmun** [comment]
Thanks for the reply, I think I have a rough idea on how to get it working. 

Currently, I am specifying dropout as a `custom_options` under `model_config` and using `tune.sample_from` to initialize the `dropout_factor`.

By doing this, I was able to access the `dropout_factor` within my model and the current `dropout_factor` for each agent shows up in the training status table. However, the `dropout_factor` does not mutate with training, the dropout for the top performing agent is copied over directly to the other agents. The `dropout_factor` is treated as a static config setting.

The next thing that I need to do is code up a `custom_explore_fn` to mutate my `dropout_factor`. I tried specifying the sampling distribution for dropout within `hyperparam_mutations` of PopulationBasedTraining for the PBT to mutate my `dropout_factor`. However, the `hyperparam_mutations` expects a dict key followed by the sampling distribution, but the dropout_factor is contained within a nested dictionary.
